.mailmap

```diff
@@ -14,163 +18,203 @@
 # Note that each qiskit element uses its own mailmap so it may be necessary to
 # propagate changes in other repos for consistency.
 #
-Abdón Rodríguez Davila <a@abdonrd.com> <a@abdonrd.com>
+Abby Mitchell <23662430+javabster@users.noreply.github.com> <abby.mitchell@btinternet.com>
+Abby Mitchell <23662430+javabster@users.noreply.github.com> <abby.s.mitchell@gmail.com>
```

Comment:
Maybe one of the actual emails (or Abby's IBM email) should be canonical here.

Reply:
If people turned on the github privacy then they presumably intended to try to reduce their spam load as intended for that feature, and any non noreply.github emails were accidents that slipped through when committing via incorrectly-configured commandlines, etc. When we first introduced the .mailmap the discussion was that we would:

1) Try fairly hard (ie sleuth a bit) to combine commits from the same individual
2) Prefer a noreply-github email address if one is present
3) Failing that prefer an ibm email address if one is present
4) Failing that prefer the globally-routed  (ie not `@macbook.local` email address that the committer used most frequently
5) Allow the committer to change this at any time by making a PR against .mailmap
6) Only do light sleuthing to associate real names with the user in case the user is trying for anonymity

This script tries to follow those rules 1-5 by sticking with the already-present email address from the mailmap and only generating a new one according to the above preference if one is absent.

I don't think there's much value in having a routable email for the mailmap - its more for tracking contributions and generating the authorlist for the DOI entry.

Reply:
Ok, if it's intentional it's fine by me.